### PR TITLE
Mistral demo improvement

### PR DESCRIPTION
**Pull request by sraizada-tt**
Added support for:
- Longer seq len (1024 tokens)
- bf4 MLP weights
- sharded rm norm 
- attention caching

fyi @mtairum @yieldthought

### CODE REVIEW

**Review by TT-billteng — CHANGES_REQUESTED** [review]
don't think you need to change taskflow for this

**Review by arakhmati — APPROVED** [review]

**Comment by mtairum** [comment]
Moved temporarily the mistral test above mamba (which was failing) just to confirm that the mistral perf tests are passing.

The logs are all good: https://github.com/tenstorrent/tt-metal/actions/runs/9005127711/job/24740023842

We'll revert the last commit, update the expected times and it will be good to merge.

**Comment by mtairum** [comment]
Reverted the taskflow (accidently committed) and the CI test order change.
Updated to latest branch. 

The CI pipeline is broken now, but not related to our changes. It's good for merge.

**Comment by mtairum** [comment]
> don't think you need to change taskflow for this

This was added by accident. Reverted on last push. @TT-billteng can you please review?

**Review by tt-rkim — APPROVED** [review]

**Review by TT-billteng — APPROVED** [review]

File: models/demos/wormhole/mistral7b/demo/demo.py [hunk 1]
```diff
@@ -126,6 +127,10 @@ def run_mistral_demo(user_input, batch_size, device):
         input_prompts, tokenizer, model_args, dtype, embd, instruct_mode, device
     )
 
+    logger.info("Caching attention ops..")
```
Comment (TT-billteng) [inline models/demos/wormhole/mistral7b/demo/demo.py:130]: is this needed?

File: models/demos/wormhole/mistral7b/tt/mistral_common.py [hunk 2]
```diff
@@ -172,3 +172,44 @@ def sample(logits: torch.Tensor, temperature: float, top_p: float):
...
+        start_pos=0,
+    )
+    for iter in range(1024):
+        pos = iter
+        tt_out = tt_model(
+            [attention_input],
+            pos,
+        )
+        # ttnn.deallocate(tt_out[0])
+        print("Cached iter", iter)
```
Comment (TT-billteng) [inline models/demos/wormhole/mistral7b/tt/mistral_common.py:205]: clean up

**Pull request merged**